File: src/main/java/tmi/recipe/types/HeatMark.java
package tmi.recipe.types;

import arc.Core;
import arc.graphics.g2d.TextureRegion;
import mindustry.gen.Icon;
import tmi.TooManyItems;

public class HeatMark extends RecipeItem<String> {
  public static HeatMark INSTANCE = TooManyItems.itemsManager.addItemWrap("heat-mark", new HeatMark());

  private HeatMark() {
    super("heat-mark");
  }

  @Override
  public int ordinal() {
    return -1;
  }

  @Override
  public int typeID() {
    return -1;
  }

  @Override
  public String name() {
    return item;
  }

  @Override
  public String localizedName() {
    return Core.bundle.get("tmi." + item);
  }

  @Override
  public TextureRegion icon() {
    return Icon.waves.getRegion();
  }

  @Override
  public boolean hidden() {
    return false;
  }
}

File: src/main/kotlin/tmi/recipe/types/HeatMark.kt
package tmi.recipe.types

import arc.Core
import arc.graphics.g2d.TextureRegion
import mindustry.gen.Icon
import tmi.recipe.AmountFormatter
import tmi.ui.SpecialFormatters
import tmi.ui.calculator.CalculatorView
import tmi.util.Consts

object HeatMark: SingleItemMark("heat-mark") {
  init { SpecialFormatters.setSpecialFormatter(this, AmountFormatter.floatFormatter()) }

  override val icon: TextureRegion get() = Icon.waves.region
  override val typeTag: String get() = Core.bundle["type.power.name"]
  override val ordinal: Int get() = 20000
  override val ownMod: String get() = Consts.VANILLA
}
